src/main/java/org/alfresco/repo/content/MimetypeMap.java

```diff
@@ -422,6 +561,71 @@ public void init()
         PropertyCheck.mandatory(this, "configService", configService);
         PropertyCheck.mandatory(this, "contentCharsetFinder", contentCharsetFinder);
 
+        if (mimetypeJsonConfigDir != null || jsonObjectMapper != null || cronExpression != null || initialAndOnErrorCronExpression != null)
```

Comment:
Do we really need this if-clause ?
From what I observe PropertyCheck.mandatory(Object target, String propertyName, Object value) verifies if an (value== null), I think we can change lines 564-570 to
```
PropertyCheck.mandatory(this, "jsonObjectMapper", jsonObjectMapper);
PropertyCheck.mandatory(this, "cronExpression", cronExpression);
PropertyCheck.mandatory(this, "initialAndOnErrorCronExpression", initialAndOnErrorCronExpression);
PropertyCheck.mandatory(this, "mimetypeJsonConfigDir", mimetypeJsonConfigDir);
```
I added mimetypeJsonConfigDir here as we verify that is not null when we read the JSON config (we can also remove that check - line 659)